mod_ci/controllers.py

```diff
@@ -905,6 +907,29 @@ def blocked_users():
             g.db.add(blocked_user)
             g.db.commit()
             flash('User blocked successfully.')
+            # Remove any queued pull request from blocked user
+            gh = GitHub(access_token=g.github['bot_token'])
+            repository = gh.repos(g.github['repository_owner'])(g.github['repository'])
+            try:
+                # Getting all pull requests by blocked user on the repo
+                pulls = requests.get('https://api.github.com/repos/ccextractor/sample-platform/pulls/head={username}'
+                                     ).format(username=usernames[addUserForm.userID.data])
+                for pull in pulls:
+                    test = Test.query.filter(Test.pr_nr == pull['number']).all()
```

Comment:
you are fetching "all" tests corresponding to a pr, You need to run a one more for loop for each test

Reply:
Ah, right.

Reply:
Fixed 😃